### ISSUE TITLE
Global mouse position is messed up when viewport is changed during runtime

### ISSUE THREAD

**Issue by Ezenki** [jocamar/Godot-Common-Game-Framework#1]
Whenever I change the size of the viewport during gameplay, the global mouse position in messed up and the engine cant calculate it correctly, is presume this is because the split-screen functionality of the code doesn't account for viewport changes during runtime.

**Comment by jocamar** [comment]
When you say changing the size of the viewport do you mean resizing the game window or changing the size property of the viewport in player_viewport.tscn?

**Comment by Ezenki** [comment]
When I resize the game window

**Comment by jocamar** [comment]
Hmm, it might be related to this issue in Godot:
https://github.com/godotengine/godot/issues/34805

I don't do anything too fancy with split-screen so I think this is probably a problem with using Viewports in Godot. Do any of the solutions in the issue above work for you?

**Comment by Ezenki** [comment]
I'm not sure, but I also just found out that using the clamp to viewport size function doesn't work.

**Comment by Ezenki** [comment]
Does the game manager have a way to get viewport rect size to use for a clamp function, because when I use the built-in get_viewport_rect().size, the character still walks past the boundary on the right and bottom parts of the screen

**Comment by jocamar** [comment]
You should be able to get the size of the viewport by calling get_viewport().get_visible_rect(). But I can add a method to get you the viewport reference so you can call any viewport methods for each viewport.

**Comment by Ezenki** [comment]
yeah, a method to call viewports would be nice, I had to work around the clamp problem by just changing the size of the player viewport scene to match my project settings.
